### DISCUSSION TITLE
Will installing the fail2ban plugin interfere with updating the main Kimai application via git?

### DISCUSSION THREAD

**Discussion by RayCulp** [kimai/kimai#3330] [On-Premise: Installation & Updates]
Won't I have to hard reset my local copy of the repo before I can check out the latest version of Kimai, thereby deleting the fail2ban plugin?

**Comment by kevinpapst** [comment]
Why? You can install the plugin like every other one. Fetch the ZIP, upload and extract it into var/plugins/ and reload the cache.

**Reply by RayCulp** [reply]
Ok, just asking. I seem to remember situations where git detected a locally changed file and wouldn't let me check out a new version of a repo until the conflict was resolved. Maybe not related. Thanks.

**Reply by kevinpapst** [reply]
That can happen, true. But not if you either use the ZIP method or do clone within the var/plugins/ directory (which is ignored by the git that is used for Kimai core).

**Reply by RayCulp** [reply]
> or do clone within the var/plugins/ directory (which is ignored by the git that is used for Kimai core)

Ah, that's the key. I got it working, by the way. It's excellent. Thank you.
